pkg/util/validation/validate.go

```diff
@@ -62,6 +73,16 @@ func (v *SchemaValidator) Validate(r io.Reader) error {
 func (v *SchemaValidator) ValidateWithMessage(r io.Reader, msg string) error {
 	node, err := v.Schema.Apply(v.Reference, r)
 	if err != nil {
+		// It is observed that json.NewDecoder.Decode and json.Unmarshal
+		// returns different error with the input is empty.
+		// json.Unmarshal returns *json.SyntaxError while
+		// json.NewDecoder.Decode returns io.EOF
+		// https://go.dev/play/p/WHEtDYzJKTo
+		// So we convert the io.EOF here.
+		if errors.Is(err, io.EOF) {
+			return errJSONSyntaxErrorOfEmptyInput
+		}
+
```

Comment:
Why we don't simply use `newInvalidJSON`?

Reply:
newInvalidJSON takes *json.SyntaxError. We do not have a *json.SyntaxError here.